python/cudnn/gemm/frost/compiler.py

```
@@ -532,25 +546,33 @@ def _smem_desc_params(
+    # no GPU is visible (render-only / CI) — keep the config's own depth then.
+    cols_per_mma_m = _epi_tile_cols(cfg, cta_group)
+    acc_region_cols = chain.num_gemms * cfg.num_mma_m * cols_per_mma_m
+    acc_stages = cfg.acc_stages
+    if total_tmem:
+        if acc_region_cols > total_tmem:
             raise NotImplementedError(
-                f"multi-GEMM: {chain.num_gemms} GEMMs × {cols_per_acc} acc cols "
-                f"= {region} exceed {total_tmem} TMEM (single stage). "
-                f"Pick a smaller cta_tile_n or fewer GEMMs."
+                f"accumulators need {acc_region_cols} TMEM columns "
+                f"({chain.num_gemms} GEMM(s) × {cfg.num_mma_m} MMA-M block(s) × "
```

Ruff (0.16.1)

[warning] String contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF001)